File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ele=nums[0];
        int count=0;
        for(int i=0;i<nums.size();i++){
            if(count==0)ele=nums[i];
            count+=(ele==nums[i])?1:-1;
        }
        return ele;
    }
};

File: 169-majority-element/majority-element.java

class Solution {
    public int majorityElement(int[] nums) {
        
        int count = 0;
        int el = nums[0];

        for(int i = 0; i<nums.length; i++){
            if(count==0){
                count++;
                el = nums[i];
            }
            else if(nums[i]==el){
                count++;
            }

            else{
                count--;
            }
        }

        return el;
        
    }
}